src/main/java/org/minions/devfund/mauricioz/Battleship/Ship.java

```diff
@@ -0,0 +1,209 @@
+    private int bowColumn;
+    private int length;
+    private boolean horizontal;
+    private boolean[] hit;
+
+    /**
+     * Getter for bowRow attribute.
+     * @return bowRow attribute.
+     */
+    public int getBowRow() {
+        return bowRow;
+    }
+
+    /**
+     * Getter for bowColumn attribute.
+     * @return bowColumn value.
+     */
+    public int getBowColumn() {
+        return bowColumn;
+    }
+
+    /**
+     * Getter for length attribute.
+     * @return length value.
+     */
+    public int getLength() {
+        return length;
+    }
+
+    /**
+     * Getter for horizontal attribute.
+     * @return horizontal value.
+     */
+    public boolean isHorizontal() {
+        return horizontal;
+    }
+
+    /**
+     * Getter for hit attribute.
+     * @return hit value.
+     */
+    public boolean[] getHit() {
+        return this.hit.clone();
+    }
+
+    /**
+     * Setter for bowRow attribute.
+     * @param bowRow value.
+     */
+    public void setBowRow(int bowRow) {
+        this.bowRow = bowRow;
+    }
+
+    /**
+     * Setter for bowColumn attribute.
+     * @param bowColumn value.
+     */
+    public void setBowColumn(int bowColumn) {
+        this.bowColumn = bowColumn;
+    }
+
+    /**
+     * Setter for length attribute.
+     * @param length value.
+     */
+    public void setLength(int length) {
+        this.length = length;
+    }
+
+    /**
+     * Setter for horizontal attribute.
+     * @param horizontal value.
+     */
+    public void setHorizontal(boolean horizontal) {
+        this.horizontal = horizontal;
+    }
+
+    /**
+     * Setter for hit attribute.
+     * @param hit value.
+     */
+    public void setHit(final boolean[] hit) {
+        this.hit = new boolean[hit.length];
+        for (int i = 0; i < hit.length; i++) {
+            this.hit[i] = hit[i];
+        }
+    }
+
+    /**
+     * Abstract method to get the ship type.
+     * @return shipType.
+     */
+    public abstract String getShipType();
+
+    /**
+     * Method to review if it is possible place a ship on a given coordinates.
+     * @param row coordinate.
+     * @param column coordinate.
+     * @param horizontal position of the ship.
+     * @param ocean object.
+     * @return true or false.
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (startOrFinishOnTheEdge(row, column, ocean) || nothingAround(row, column, horizontal, ocean)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * verify is not another ship near of the desired position.
+     * @param row value.
+     * @param column value.
+     * @param horizontal ship direction.
+     * @param ocean board.
+     * @return true or false.
+     */
+    private boolean nothingAround(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (horizontal && nothingAroundForPlacedHorizontal(row, column, ocean)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)